packages/terra-time-input/translations/es.json

```diff
@@ -1,6 +1,6 @@
 {
-  "Terra.timeInput.am": "a. m.",
-  "Terra.timeInput.pm": "p. m.",
+  "Terra.timeInput.am": "",
+  "Terra.timeInput.pm": "",
```

Comment:
The [current logic](https://github.com/cerner/terra-framework/blob/main/packages/terra-time-input/src/TimeUtil.js#L262-L271) that relies on the existence of the Terra.timeInput.am string to determine the variant is confusing and not obvious. We should clearly define in a list which locale can support the 12-hour notation and check against this list. If a locale is not in this list, then it should only support the 24-hour notation.

These are the locales that have a non-empty string for `Terra.timeInput.am`/`Terra.timeInput.pm` so they are the ones that can support the 12-hour notation.

- en-AU
- en-CA
- en-GB
- en-US
- en
- es-US

Reply:
Updated 4597abe